### PR TITLE
feat: add cis aws 5.1 rule

### PR DESCRIPTION
**Pull request by olegsu**
### Summary of your changes
Add CIS AWS networking rule 

### Related Issues
- https://github.com/elastic/csp-security-policies/issues/149
- https://github.com/elastic/cloudbeat/pull/595
- closes: https://github.com/elastic/csp-security-policies/issues/149
### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by oren-zohar** [comment]
Please run [`generate_rule_metadata`](https://github.com/elastic/csp-security-policies/tree/main/dev#generate_rule_metadatapy) :pray:

File: bundle/compliance/lib/common/common.rego [hunk 1]
```diff
@@ -5,6 +5,10 @@ metadata = {"opa_version": opa_version}
 # get OPA version
 opa_version := opa.runtime().version
 
+has_key(x, k) {
+	x[k]
+}
+
```
Comment (oren-zohar) [inline bundle/compliance/lib/common/common.rego:11]: we can keep it as it makes the check clearer, but checking if key exists using `x[k]` is a basic feature in rego

File: bundle/compliance/policy/aws_ec2/ensure_public_ingress.rego [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	# filter all the entries that 
+	# 1. have ingres (egress == false)
+	# 2. allow any source ip of 0.0.0.0/0
+	entries := [entry | entry := data_adapter.nacl_entries[_]; entry.Egress == false; entry.CidrBlock == "0.0.0.0/0"; entry.RuleAction == "allow"]
+
+	# # make sure that all the matched entries on the filter does not expose any port that is less than 1024
+	# # such as SSH (22) or RDP (3389)
+	every entry in entries {
+		not common.has_key(entry, "PortRange")
+		entry.PortRange.From <= 1024
```
Comment (oren-zohar) [inline bundle/compliance/policy/aws_ec2/ensure_public_ingress.rego:31]: maybe I'm missing something, but wouldn't `3389` will pass this check?

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: bundle/compliance/policy/aws_ec2/ensure_public_ingress.rego (hunk 2 above)
Comment (olegsu) [reply bundle/compliance/policy/aws_ec2/ensure_public_ingress.rego:31]: From CIS 
> A Port value of ALL or a port range such as 0-1024 are inclusive of port 22, 3389, and other remote server administration ports

But you are right that the check is not good enough in that context.
I will add also checks for specific ports

Comment (oren-zohar) [reply bundle/compliance/policy/aws_ec2/ensure_public_ingress.rego:31]: I saw it also, but it doesn't make sense? what am I missing?

File: bundle/compliance/lib/common/common.rego (hunk 1 above)
Comment (olegsu) [reply bundle/compliance/lib/common/common.rego]: removed

File: bundle/compliance/policy/aws_ec2/ensure_public_ingress.rego (hunk 2 above)
Comment (olegsu) [reply bundle/compliance/policy/aws_ec2/ensure_public_ingress.rego]: you are right, it does not make any sense.
removing

File: bundle/compliance/policy/aws_ec2/ports_test.rego [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package compliance.policy.aws_ec2.ports
+
+test_pass {
+	in_range(80, 443, 443) == true
+	in_range(80, 443, 80) == true
+	in_range(80, 443, 100) == true
```
Comment (oren-zohar) [inline bundle/compliance/policy/aws_ec2/ports_test.rego:6]: nit: in rego it's common to leave it implicit
```suggestion
	in_range(80, 443, 443)
	in_range(80, 443, 80)
	in_range(80, 443, 100)
```

File: bundle/compliance/policy/aws_ec2/ports_test.rego [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+test_fail {
+	in_range(80, 443, 444) == false
+	in_range(80, 443, 30) == false
```
Comment (oren-zohar) [inline bundle/compliance/policy/aws_ec2/ports_test.rego:11]: nit: more rego conventions 
```suggestion
	not in_range(80, 443, 444)
	not in_range(80, 443, 30)
```

File: bundle/compliance/policy/aws_ec2/ports.rego [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package compliance.policy.aws_ec2.ports
+
+# Admin ports are network ports that are reserved for use by system administrators to manage servers and other network devices.
+# These ports are typically used for remote management, monitoring, and control of devices over a network
+admin_ports = {22, 23, 25, 53, 80, 110, 143, 389, 443, 465, 587, 636, 993, 995, 3389}
+
+# check whether a given value (candidate) is within a range of values specified by from and to
+in_range(from, to, candidate) {
+	print(from, to, candidate, candidate >= from, candidate < to)
```
Comment (oren-zohar) [inline bundle/compliance/policy/aws_ec2/ports.rego:9]: did you leave the print by design here?

**Review by oren-zohar — CHANGES_REQUESTED** [review]

**Review by oren-zohar — APPROVED** [review]
LGTM :rocket:

**Pull request merged**